### DISCUSSION TITLE
Get Result from CommandTestFactory.run

### DISCUSSION THREAD

**Discussion by ColiZei** [jmcdo29/nest-commander#725] [Q&A] [answered]
Hey!

I would like to write some e2e tests and check the expected result like a response body.

Is it possible?
Any help?

Example:

```ts

import { TestingModule } from "@nestjs/testing";
import { CommandTestFactory } from "nest-commander-testing";
import { AppModule } from "../../../src/app.module";

describe("Health Check Cronjob (e2e)", () => {
  let commandInstance: TestingModule;

  beforeAll(async () => {
    commandInstance = await CommandTestFactory.createTestingCommand({
      imports: [AppModule],
    }).compile();
  });

  it("Should run successful", async () => {
    const result = await CommandTestFactory.run(commandInstance, [
      "health-check-cronjob",
    ]);

    expect(result.status).toBe("ok");
  });

  afterAll(async () => {
    await commandInstance.close();
  });
});


```

Thanks for your help!

**Comment by jmcdo29** [comment, marked as answer]
Currently, there isn't a way to return the result of the command, mainly due to the fact that when running the command there's no return to anything, usually only outcomes like writing a log to the terminal, making changes to a database, or calling an API. 

I could probably write up a way to get the `CommandTestFactory.run()` to return whatever the result of whatever command ran, to help with testing, but you'd almost be better off spying on the command and its `run` method and asserting what the spy responds with.

**Reply by ColiZei** [reply]
Thanks for your answer!

Maybe i do something wrong, lag of experience or don't get into it throught your example in the documentation.

I updated the code:

```ts
import { TestingModule } from "@nestjs/testing";
import * as childProcess from "child_process";
import { CommandTestFactory } from "nest-commander-testing";
import { AppModule } from "../../../src/app.module";

describe("Health Check Cronjob (e2e)", () => {
  let commandInstance: TestingModule;

  beforeAll(async () => {
    commandInstance = await CommandTestFactory.createTestingCommand({
      imports: [AppModule],
    }).compile();
  });

  it("Should run successful", async () => {
    const spawnSpy = jest.spyOn(childProcess, "spawn");
    await CommandTestFactory.run(commandInstance, ["health-check-cronjob"]);

    expect(spawnSpy).toBeCalled();
  });

  afterAll(async () => {
    await commandInstance.close();
  });
});
```

Result:

```sh
  Health Check Cronjob (e2e)
    ✕ Should run successful (38 ms)

  ● Health Check Cronjob (e2e) › Should run successful

    expect(jest.fn()).toBeCalled()

    Expected number of calls: >= 1
    Received number of calls:    0

      17 |     await CommandTestFactory.run(commandInstance, ["health-check-cronjob"]);
      18 |
    > 19 |     expect(spawnSpy).toBeCalled();
         |                      ^
      20 |   });
      21 |
      22 |   afterAll(async () => {

      at Object.<anonymous> (test/cronjobs/health/health.e2e-spec.ts:19:22)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        6.729 s
Ran all test suites matching /test\/cronjobs\/health\/health.e2e-spec.ts/i.
```

Any Idea?

**Reply by jmcdo29** [reply]
Can you show the command you're trying to test?

**Reply by ColiZei** [reply]
I got it!

It was my mistake. Did something wrong, sry.

Thy for your help and this helpfull project!

**Reply by ColiZei** [reply]
I created a pull request for another feater i need for e2e testing.
https://github.com/jmcdo29/nest-commander/pull/731
Thank you.

**Reply by jharrisonio** [reply]
> I got it!
> 
> It was my mistake. Did something wrong, sry.
> 
> Thy for your help and this helpfull project!

@ColiZei what was your solution? I'm having the same problem.

**Reply by TomOthentic** [reply]
I'm also curious as to how you spy on it @ColiZei

**Reply by ColiZei** [reply]
I had no solution. This was what i needed https://github.com/jmcdo29/nest-commander/pull/731
